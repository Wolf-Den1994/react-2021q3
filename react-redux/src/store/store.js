import { createStore, combineReducers } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import { sortReducer } from './sortReducer';
import { searchStringReducer } from './searchStringReducer';
import { numberResultReducer } from './numberResultReducer';

const rootReducer = combineReducers({
  sort: sortReducer,
  searchString: searchStringReducer,
  numberResult: numberResultReducer,
});

const store = createStore(rootReducer, composeWithDevTools());

export default store;
