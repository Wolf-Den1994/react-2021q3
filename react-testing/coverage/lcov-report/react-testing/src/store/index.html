
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for react-testing/src/store</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../index.html">All files</a> react-testing/src/store</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">54.81% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>57/104</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">37.5% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>12/32</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">27.59% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>8/29</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">63.64% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>56/88</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line medium'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="controlReducer.js"><a href="controlReducer.js.html">controlReducer.js</a></td>
	<td data-value="58.33" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 58%"></div><div class="cover-empty" style="width: 42%"></div></div>
	</td>
	<td data-value="58.33" class="pct medium">58.33%</td>
	<td data-value="24" class="abs medium">14/24</td>
	<td data-value="28.57" class="pct low">28.57%</td>
	<td data-value="7" class="abs low">2/7</td>
	<td data-value="16.67" class="pct low">16.67%</td>
	<td data-value="6" class="abs low">1/6</td>
	<td data-value="73.68" class="pct medium">73.68%</td>
	<td data-value="19" class="abs medium">14/19</td>
	</tr>

<tr>
	<td class="file low" data-value="detailsReducer.js"><a href="detailsReducer.js.html">detailsReducer.js</a></td>
	<td data-value="33.33" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 33%"></div><div class="cover-empty" style="width: 67%"></div></div>
	</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="24" class="abs low">8/24</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="6" class="abs low">2/6</td>
	<td data-value="11.11" class="pct low">11.11%</td>
	<td data-value="9" class="abs low">1/9</td>
	<td data-value="38.1" class="pct low">38.1%</td>
	<td data-value="21" class="abs low">8/21</td>
	</tr>

<tr>
	<td class="file medium" data-value="homeReducer.js"><a href="homeReducer.js.html">homeReducer.js</a></td>
	<td data-value="60.53" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 60%"></div><div class="cover-empty" style="width: 40%"></div></div>
	</td>
	<td data-value="60.53" class="pct medium">60.53%</td>
	<td data-value="38" class="abs medium">23/38</td>
	<td data-value="42.86" class="pct low">42.86%</td>
	<td data-value="14" class="abs low">6/14</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="10" class="abs medium">5/10</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="33" class="abs medium">22/33</td>
	</tr>

<tr>
	<td class="file medium" data-value="searchBarReducer.js"><a href="searchBarReducer.js.html">searchBarReducer.js</a></td>
	<td data-value="62.5" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 62%"></div><div class="cover-empty" style="width: 38%"></div></div>
	</td>
	<td data-value="62.5" class="pct medium">62.5%</td>
	<td data-value="16" class="abs medium">10/16</td>
	<td data-value="40" class="pct low">40%</td>
	<td data-value="5" class="abs low">2/5</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="4" class="abs low">1/4</td>
	<td data-value="76.92" class="pct medium">76.92%</td>
	<td data-value="13" class="abs medium">10/13</td>
	</tr>

<tr>
	<td class="file high" data-value="store.js"><a href="store.js.html">store.js</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Thu Aug 19 2021 16:45:27 GMT+0300 (GMT+03:00)
            </div>
        </div>
        <script src="../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../sorter.js"></script>
        <script src="../../../block-navigation.js"></script>
    </body>
</html>
    