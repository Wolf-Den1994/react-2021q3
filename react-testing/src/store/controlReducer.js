export const defaultState = {
  sort: 'popularity',
  page: 1,
  numberResult: 10,
  disabledBtn: [true, false],
  activeBtnSort: [false, true, false],
};

export const CHANGE_SORT = 'CHANGE_SORT';
export const CHANGE_PAGE_ON_HOME = 'CHANGE_PAGE_ON_HOME';
export const CHANGE_NUMBER_RESULT = 'CHANGE_NUMBER_RESULT';
export const CHANGE_DISABLED_BTN = 'CHANGE_DISABLED_BTN';
export const CHANGE_ACTIVE_BTN_SORT = 'CHANGE_ACTIVE_BTN_SORT';

export const controlReducer = (state = defaultState, action) => {
  switch (action.type) {
    case CHANGE_SORT:
      return { ...state, sort: action.payload };

    case CHANGE_PAGE_ON_HOME:
      return { ...state, page: action.payload };

    case CHANGE_NUMBER_RESULT:
      return { ...state, numberResult: action.payload };

    case CHANGE_DISABLED_BTN:
      return { ...state, disabledBtn: action.payload };

    case CHANGE_ACTIVE_BTN_SORT:
      return { ...state, activeBtnSort: action.payload };

    default:
      return state;
  }
};

export const changeSortAction = (payload) => ({
  type: CHANGE_SORT,
  payload,
});

export const changePageAction = (payload) => ({
  type: CHANGE_PAGE_ON_HOME,
  payload,
});

export const changeNumberResultAction = (payload) => ({
  type: CHANGE_NUMBER_RESULT,
  payload,
});

export const changeDisabledBtnAction = (payload) => ({
  type: CHANGE_DISABLED_BTN,
  payload,
});

export const changeActiveBtnSortAction = (payload) => ({
  type: CHANGE_ACTIVE_BTN_SORT,
  payload,
});
