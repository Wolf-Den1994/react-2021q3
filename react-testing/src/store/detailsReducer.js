import NewsServise from '../services/new-service';

export const defaultState = {
  data: {},
  loading: true,
};

export const RECEIVING_DETAILS_DATA_FROM_SERVER =
  'RECEIVING_DETAILS_DATA_FROM_SERVER';
export const CHANGE_LOADING_DETAILS = 'CHANGE_LOADING_DETAILS';

export const detailsReducer = (state = defaultState, action) => {
  switch (action.type) {
    case RECEIVING_DETAILS_DATA_FROM_SERVER:
      return { ...state, data: action.payload };

    case CHANGE_LOADING_DETAILS:
      return { ...state, loading: action.payload };

    default:
      return state;
  }
};

export const changeDetailsDataAction = (payload) => ({
  type: RECEIVING_DETAILS_DATA_FROM_SERVER,
  payload,
});

export const changeLoadingDetailsAction = (payload) => ({
  type: CHANGE_LOADING_DETAILS,
  payload,
});

export function fetchDataForDetails(q, sort, id) {
  const data = (dispatch) => {
    const newsServise = new NewsServise();
    dispatch(changeLoadingDetailsAction(true));
    newsServise
      .getResourseById(q, sort)
      .then((response) => {
        if (Object.prototype.hasOwnProperty.call(response, 'articles')) {
          const result = response.articles.find((el, index) => +id === index);
          const obj = {
            articles: [result],
          };
          dispatch(changeDetailsDataAction(obj));
        }
      })
      .catch((e) => console.error('Error: ', e))
      .finally(() => {
        dispatch(changeLoadingDetailsAction(false));
      });
  };
  return data;
}
