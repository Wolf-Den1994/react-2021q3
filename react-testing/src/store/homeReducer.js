import NewsServise from '../services/new-service';

export const defaultState = {
  data: {},
  totalResults: null,
  loading: false,
  empty: false,
};

export const RECEIVING_DATA_FROM_SERVER = 'RECEIVING_DATA_FROM_SERVER';
export const CHANGE_TOTAL_RESULT = 'CHANGE_TOTAL_RESULT';
export const CHANGE_LOADING = 'CHANGE_LOADING';
export const CHANGE_EMPTY = 'CHANGE_EMPTY';

export const homeReducer = (state = defaultState, action) => {
  switch (action.type) {
    case RECEIVING_DATA_FROM_SERVER:
      return { ...state, data: action.payload };

    case CHANGE_TOTAL_RESULT:
      return { ...state, totalResults: action.payload };

    case CHANGE_LOADING:
      return { ...state, loading: action.payload };

    case CHANGE_EMPTY:
      return { ...state, empty: action.payload };

    default:
      return state;
  }
};

export const changeDataAction = (payload) => ({
  type: RECEIVING_DATA_FROM_SERVER,
  payload,
});

export const changeTotalResultsAction = (payload) => ({
  type: CHANGE_TOTAL_RESULT,
  payload,
});

export const changeLoadingAction = (payload) => ({
  type: CHANGE_LOADING,
  payload,
});

export const changeEmptyAction = (payload) => ({
  type: CHANGE_EMPTY,
  payload,
});

export const fetchDataForHome = (searchString, sort, numberResult, page) => {
  const data = (dispatch) => {
    const newsServise = new NewsServise();
    return newsServise
      .getResourse(searchString, sort, numberResult, page)
      .then((response) => {
        if (Object.prototype.hasOwnProperty.call(response, 'articles')) {
          const addId = response.articles.map((el, index) => {
            return { ...el, id: index + numberResult * (page - 1) };
          });
          const objData = {
            articles: addId,
          };
          dispatch(changeDataAction(objData));
        }
        dispatch(changeTotalResultsAction(response.totalResults));
        if (Object.keys(response).length) dispatch(changeLoadingAction(false));
        if (Object.prototype.hasOwnProperty.call(response, 'articles')) {
          if (Object.keys(response.articles).length) {
            dispatch(changeEmptyAction(true));
          } else {
            dispatch(changeEmptyAction(false));
          }
        }
      })
      .catch((e) => console.error('Error: ', e));
  };
  return data;
};
