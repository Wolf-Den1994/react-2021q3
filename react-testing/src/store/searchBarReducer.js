export const defaultState = {
  valueSearchBar: '',
  searchString: '',
  disabledBtnSearchBar: true,
};

export const CHANGE_VALUE_SEARCHBAR = 'CHANGE_VALUE_SEARCHBAR';
export const CHANGE_SEARCH = 'CHANGE_SEARCH';
export const CHANGE_DISABLED_BTN_SEARCHBAR = 'CHANGE_DISABLED_BTN_SEARCHBAR';

export const searchBarReducer = (state = defaultState, action) => {
  switch (action.type) {
    case CHANGE_VALUE_SEARCHBAR:
      return { ...state, valueSearchBar: action.payload };

    case CHANGE_SEARCH:
      return { ...state, searchString: action.payload };

    case CHANGE_DISABLED_BTN_SEARCHBAR:
      return { ...state, disabledBtnSearchBar: action.payload };

    default:
      return state;
  }
};

export const changeValueSearchBarAction = (payload) => ({
  type: CHANGE_VALUE_SEARCHBAR,
  payload,
});

export const changeSearchStringAction = (payload) => ({
  type: CHANGE_SEARCH,
  payload,
});

export const changeDisabledBtnSearchBarAction = (payload) => ({
  type: CHANGE_DISABLED_BTN_SEARCHBAR,
  payload,
});
