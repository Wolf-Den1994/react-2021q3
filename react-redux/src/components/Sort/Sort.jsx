import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { changeSortAction } from '../../store/sortReducer';
import './Sort.scss';

const Sort = () => {
  const dispatch = useDispatch();

  const [activeBtn, setActiveBtn] = useState([false, true, false]);

  const changeSort = (event) => {
    const { target } = event;
    setActiveBtn((store) => {
      const copy = [...store];
      const falsesStore = copy.map(() => false);
      const newStore = falsesStore.map((elem, index) => {
        if (+target.classList[1] === index) {
          return true;
        }
        return elem;
      });
      return newStore;
    });
    dispatch(changeSortAction(target.id));
    target.classList.add('active');
  };

  const onClickBtns = (event) => changeSort(event);
  const onKeyClickBtns = (event) => {
    if (event.key !== 'Enter') return;
    changeSort(event);
  };

  return (
    <div
      className="sort-wrapper"
      onClick={onClickBtns}
      onKeyDown={onKeyClickBtns}
      role="presentation"
    >
      <button
        className={`sort-btn 0 ${activeBtn[0] ? 'active' : ''}`}
        id="relevancy"
        type="button"
      >
        Relevancy
      </button>
      <button
        className={`sort-btn 1 ${activeBtn[1] ? 'active' : ''}`}
        id="popularity"
        type="button"
      >
        Popularity
      </button>
      <button
        className={`sort-btn 2 ${activeBtn[2] ? 'active' : ''}`}
        id="publishedAt"
        type="button"
      >
        Published at
      </button>
    </div>
  );
};

export default Sort;
