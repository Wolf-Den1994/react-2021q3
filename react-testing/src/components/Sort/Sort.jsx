import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {
  changeSortAction,
  changeActiveBtnSortAction,
} from '../../store/controlReducer';
import './Sort.scss';

const Sort = () => {
  const dispatch = useDispatch();
  const activeBtn = useSelector((state) => state.control.activeBtnSort);

  const changeSort = (event) => {
    const { target } = event;
    const setActiveBtn = (store) => {
      const copy = [...store];
      const falsesStore = copy.map(() => false);
      const newStore = falsesStore.map((elem, index) => {
        if (+target.classList[1] === index) {
          return true;
        }
        return elem;
      });
      return newStore;
    };
    dispatch(changeActiveBtnSortAction(setActiveBtn(activeBtn)));
    dispatch(changeSortAction(target.id));
    target.classList.add('active');
  };

  const onClickBtns = (event) => changeSort(event);
  const onKeyClickBtns = (event) => {
    if (event.key !== 'Enter') return;
    changeSort(event);
  };

  return (
    <div
      className="sort-wrapper"
      onClick={onClickBtns}
      onKeyDown={onKeyClickBtns}
      role="presentation"
    >
      <button
        data-testid="relevancy"
        className={`sort-btn 0 ${activeBtn[0] ? 'active' : ''}`}
        id="relevancy"
        type="button"
      >
        Relevancy
      </button>
      <button
        data-testid="popularity"
        className={`sort-btn 1 ${activeBtn[1] ? 'active' : ''}`}
        id="popularity"
        type="button"
      >
        Popularity
      </button>
      <button
        data-testid="publishedAt"
        className={`sort-btn 2 ${activeBtn[2] ? 'active' : ''}`}
        id="publishedAt"
        type="button"
      >
        Published at
      </button>
    </div>
  );
};

export default Sort;
