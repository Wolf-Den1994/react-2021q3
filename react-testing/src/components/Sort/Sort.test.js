/* eslint-disable no-undef */
import React from 'react';
import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import store from '../../store/store';
import '@testing-library/jest-dom/extend-expect';
import Sort from './Sort';

describe('Sort component', () => {
  it('render without crashing', () => {
    render(
      // eslint-disable-next-line react/jsx-filename-extension
      <Provider store={store}>
        <Sort />
      </Provider>,
    );

    const leftClick = { button: 0 };
    const btnRelevancy = screen.getByTestId(/relevancy/i);
    const btnPopularity = screen.getByTestId(/popularity/i);
    const btnPublished = screen.getByTestId(/publishedAt/i);

    userEvent.click(btnRelevancy, leftClick);
    userEvent.click(btnPopularity, leftClick);
    userEvent.click(btnPublished, leftClick);

    userEvent.keyboard('{enter}');
    userEvent.keyboard('{esc}');
  });

  it('matches snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Sort />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
