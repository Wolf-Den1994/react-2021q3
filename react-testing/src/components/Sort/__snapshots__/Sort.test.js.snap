// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort component matches snapshot 1`] = `
<div
  className="sort-wrapper"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="presentation"
>
  <button
    className="sort-btn 0 "
    data-testid="relevancy"
    id="relevancy"
    type="button"
  >
    Relevancy
  </button>
  <button
    className="sort-btn 1 "
    data-testid="popularity"
    id="popularity"
    type="button"
  >
    Popularity
  </button>
  <button
    className="sort-btn 2 active"
    data-testid="publishedAt"
    id="publishedAt"
    type="button"
  >
    Published at
  </button>
</div>
`;
